include/envoy/http/filter.h

```diff
@@ -283,6 +283,17 @@ class StreamFilterCallbacks {
    * @return the ScopeTrackedObject for this stream.
    */
   virtual const ScopeTrackedObject& scope() PURE;
+
+  /**
+   * Should be used when a filter's callback is executed, and we continue
+   * processing the request. Some crash context that is lost when we resume the
+   * filter chain from commonContinue(), can be restored with this.
+   *
+   * @return an opaque ScopeTrackedObject that will restore context when
+   * continuing.
+   */
+  // TODO(kbaichoo): better naming and commenting
+  virtual const ScopeTrackedObject& restoreContextOnContinue() PURE;
```

Comment:
There's the potential question of the resumption being in the request or response direction.  The example we have seen so far involves resumption in request processing, but I think it's also possible to get resumption in the response direction.  Would be good to make this method name capture that in some way, possibly by including "ForEncode" / "ForDecode" in the name or possibly something about request/response.  I don't know which is appropriate (if any) without looking more closely at the implementation of StreamFilterCallbacks.

Reply:
That's a good point, I will expand upon that. 

I don't know of resumption in the response direction, but I could see some use cases.

Reply:
After looking at it a bit more, I think there can be one unified method. For both cases this will end up adding the L4 (between downstream and Envoy) and L7 info in the crash